src/Servers/Kestrel/Core/src/Internal/Certificates/CertificateConfigLoader.cs

```diff
@@ -57,10 +70,10 @@ public X509Certificate2 LoadCertificate(CertificateConfig certInfo, string endpo
                     {
                         if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
                         {
-                            return PersistKey(certificate);
+                            return (PersistKey(certificate), intermediates);
                         }
 
-                        return certificate;
+                        return (certificate, null);
```

Comment:
Why would you ignore the intermediates on non-Windows when a key.pem is provided?

Reply:
Good point, this is file based certs so I shouldn't special case Windows. I was thinking we shouldn't mix file based and store based certs, but that isn't the case here

Reply:
Ah it was a bug, see this is why I need tests 😄